producer-kafka/src/test/java/io/stargate/producer/kafka/IntegrationTestBase.java

```diff
@@ -0,0 +1,167 @@
+
+  protected static final String TOPIC_PREFIX = "topicPrefix";
+  protected static final String PARTITION_KEY_VALUE = "pk_value";
+  protected static final Integer CLUSTERING_KEY_VALUE = 1;
+
+  @BeforeAll
+  public static void setup() throws Exception {
+    Network network = Network.newNetwork();
+    kafkaContainer = new KafkaContainer().withNetwork(network).withEmbeddedZookeeper();
+    kafkaContainer.start();
+    try (ServerSocket serverSocket = new ServerSocket(0)) {
+
+      schemaRegistry =
+          new EmbeddedSchemaRegistryServer(
+              String.format("http://localhost:%s", serverSocket.getLocalPort()),
+              String.format("localhost:%s", ZOOKEEPER_PORT),
+              kafkaContainer.getBootstrapServers());
+    }
+    schemaRegistry.startSchemaRegistry();
+  }
+
+  @AfterAll
+  public static void cleanup() {
+    kafkaContainer.stop();
+    schemaRegistry.close();
+  }
+
+  protected TableMetadata mockTableMetadata() {
+    TableMetadata tableMetadata = mock(TableMetadata.class);
+    when(tableMetadata.getKeyspace()).thenReturn("keyspaceName");
+    when(tableMetadata.getName()).thenReturn("tableName" + UUID.randomUUID().toString());
+    return tableMetadata;
+  }
+
+  protected String creteTopicName(TableMetadata tableMetadata) {
+    return String.format(
+        "%s.%s.%s", TOPIC_PREFIX, tableMetadata.getKeyspace(), tableMetadata.getName());
+  }
+
+  @NotNull
+  protected Map<String, Object> createKafkaProducerSettings(Map<String, Object> overrides) {
+    Map<String, Object> properties = new HashMap<>();
+    properties.put(CDC_TOPIC_PREFIX_NAME, TOPIC_PREFIX);
+    // metrics disabled in tests by default
+    properties.put(METRICS_ENABLED_SETTING_NAME, false);
+
+    properties.put(
+        withCDCPrefixPrefix(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG),
+        kafkaContainer.getBootstrapServers());
+    properties.put(
+        withCDCPrefixPrefix(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG), KafkaAvroSerializer.class);
+    properties.put(
+        withCDCPrefixPrefix(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG),
+        KafkaAvroSerializer.class);
+    // lower the max.block to allow faster failure scenario testing
+    properties.put(withCDCPrefixPrefix(ProducerConfig.MAX_BLOCK_MS_CONFIG), "2000");
+
+    properties.put(
+        withCDCPrefixPrefix("schema.registry.url"), schemaRegistry.getSchemaRegistryUrl());
+
+    for (Map.Entry<String, Object> override : overrides.entrySet()) {
+      properties.put(override.getKey(), override.getValue());
+    }
+    return properties;
+  }
+
+  @NotNull
+  protected Map<String, Object> createKafkaProducerSettings() {
+    return createKafkaProducerSettings(Collections.emptyMap());
+  }
+
+  @NotNull
+  private String withCDCPrefixPrefix(String settingName) {
+    return String.format("%s.%s", ConfigLoader.CDC_KAFKA_PRODUCER_SETTING_PREFIX, settingName);
+  }
+
+  protected void validateThatWasSendToKafka(
```

Comment:
Minor grammatical nit
```suggestion
  protected void verifyReceivedByKafka(
```